src/main/java/com/thirdcc/webapp/service/impl/EventAttendeeServiceImpl.java

```diff
@@ -42,9 +51,23 @@ public EventAttendeeServiceImpl(EventAttendeeRepository eventAttendeeRepository,
     @Override
     public EventAttendeeDTO save(EventAttendeeDTO eventAttendeeDTO) {
         log.debug("Request to save EventAttendee : {}", eventAttendeeDTO);
-        EventAttendee eventAttendee = eventAttendeeMapper.toEntity(eventAttendeeDTO);
-        eventAttendee = eventAttendeeRepository.save(eventAttendee);
-        return eventAttendeeMapper.toDto(eventAttendee);
+
+        Set<EventStatus> eventStatuses = new HashSet<EventStatus>() {{
+            add(EventStatus.OPEN);
+            add(EventStatus.POSTPONED);
+        }};
+
+        Event event = eventRepository
+            .findOneByIdAndStatusIn(eventAttendeeDTO.getEventId(), eventStatuses)
+            .get();
+
+        if(event != null){
+            EventAttendee eventAttendee = eventAttendeeMapper.toEntity(eventAttendeeDTO);
+            eventAttendee = eventAttendeeRepository.save(eventAttendee);
+            return eventAttendeeMapper.toDto(eventAttendee);
+        } else {
+            return new BadRequestException("This event does not exists or it is not happening"));
```

Comment:
use throw instead of return when you want to throw an error/exception

Reply:
Noted